src/dataflow/src/render/debezium.rs

```diff
@@ -118,6 +124,273 @@ where
+                        },
+                    ..
+                },
+            ..
+        } => (
+            *transaction_idx,
+            tx_id_idx.expect("render_tx should only be called when there's a transaction"),
+        ),
+        DebeziumEnvelope {
+            mode: DebeziumMode::None,
+            ..
+        } => panic!("render_tx should be called with a dedup mode"),
+    };
+
+    // XXX(chae): Should we try to avoid moving everything to one worker?
+    tx_ok
+        .inner
+        .binary_frontier(
+            &input,
+            Exchange::new(move |_| hashed_id),
+            Exchange::new(move |_| hashed_id),
+            "envelope-debezium-tx",
+            {
+                let envelope = envelope.clone();
+                let mut tx_data = vec![];
+                let mut data = vec![];
+                let mut data_buffer = VecDeque::new();
+                let mut dedup_state = HashMap::new();
+                let mut tx_mapping = HashMap::new();
+                let mut tx_event_count = HashMap::new();
+                let mut tx_cap_map = HashMap::new();
+                move |mut binary_cap, op_info| {
+                    let activator = input.scope().activator_for(&op_info.address[..]);
+                    move |tx_metadata_input, data_input, output| {
+                        while let Some((tx_metadata_cap, refmut_data)) = tx_metadata_input.next() {
+                            refmut_data.swap(&mut tx_data);
+                            let tx_metadata_cap = Rc::new(tx_metadata_cap.retain());
+                            for (row, time, diff) in tx_data.drain(..) {
+                                assert_eq!(diff, 1);
+                                let mut i = row.iter();
+                                let status = i.next().expect("status field validated to exist").unwrap_str();
+                                let tx_id = i.next().expect("tx_id field validated exist").unwrap_str().to_owned();
+                                let event_count: Option<i64> =
+                                    match i.next().expect("event_count field validated exist") {
+                                        Datum::Int16(i) => Some(i.into()),
+                                        Datum::Int32(i) => Some(i.into()),
+                                        Datum::Int64(i) => Some(i),
+                                        Datum::Null => None,
+                                        d => panic!("event_count field previously validated to be integer type.  Found {:?}", d),
+                                    };
+                                if status != "END" {
+                                    continue;
+                                }
+                                let event_count = match event_count {
+                                    Some(c) => c,
+                                    None => {
+                                        // XXX(chae): maybe a panic??
```

Comment:
Yes I think this is the correct behaviour. We should probably flip a flag so that the operator permanently enters an errored state in which the only thing it does is empty its inputs and drop them on the floor